tests/Doctrine/Tests/DBAL/Functional/ResultCacheTest.php

```diff
@@ -238,25 +238,25 @@ public function testFetchingAllRowsSavesCache(callable $fetchAll): void
     public static function fetchAllProvider(): iterable
     {
         yield 'fetchAll' => [
-            static function (ResultCacheStatement $statement): void {
+            static function (ForwardCompatibility\Result $statement): void {
```

Comment:
The expected `ResultCacheStatement` type was unnecessarily specific. Let's loosen it to `Driver\ResultStatement`. If it had been implemented this way, it wouldn't require a change now.